src/inc/com.h

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
```

Comment:
Where did this implementation come from?

Reply:
It's an amalgamation of:
https://github.com/dotnet/runtime/blob/release/6.0/src/coreclr/interop/comwrappers.hpp#L229
https://github.com/dotnet/runtime/blob/release/6.0/src/tests/profiler/native/holder.h#L7
if I recall correctly

Reply:
And https://github.com/microsoft/CLRInstrumentationEngine/blob/main/src/unix/inc/atl/atlcomcli.h#L638 for the macros.